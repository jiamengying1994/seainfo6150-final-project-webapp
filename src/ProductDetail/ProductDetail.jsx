import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';
import styles from './ProductDetail.module.css';

const ProductDetail = ({ product}) => {




    return (
      <div className={styles.ProductDetail}>

        <div className={styles.detailbox}>
          <div className={styles.detailImage}><img src={product.url} alt={product.category}/></div>
          <div className={styles.detailList}>
             <h1> {product.title} </h1>
            <ul>
              <li><span>{product.category}</span><p>Category</p></li>
              <li><span>{product.Dimensions}</span><p>Dimensions</p></li>
              <li><span>{product.Techniques}</span><p>Techniques</p></li>
            </ul>
            <div className={styles.description}>{product.text}</div>
            {/* start order button */}

          {/* end order button */}
          </div>

        </div>
      </div>
    );
  }



export default ProductDetail;