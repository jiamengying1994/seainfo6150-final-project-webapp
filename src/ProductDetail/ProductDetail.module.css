.detailbox {
    display: grid;
    grid-template-columns: 45% 45%;
    grid-template-areas: "left right";
}
.detailList {
    grid-area: right;
}

.detailList ul {
    text-align: left;
}
.detailImage {
    grid-area: left;
}

.detailList  h1 {
  position: relative;
  margin-left: 1em;
}

.detailImage img {
    width: 90%;
    z-index: -1;
}

.detailList li {
    position: relative;
    list-style: none;
    display: inline-block;
    border-right: 1px solid #ededed;
    line-height: 150%;
    color: #333333;
    text-align: center;
    padding: 0 3rem;
}

.detailList span {
    font-weight: bolder;
    font-size: 1rem;
}

.description {
    position: relative;
    padding: 5px 24px;
    margin-left: 1em;
    height: 5rem;
    line-height: 2rem;
    text-align: left;
    font-size: 1rem;
}



@media(max-width: 900px) {
.detailList  h1 {
  font-size: 1.5rem;

}
.detailList li {
    font-size: 0.5rem;

}

.detailList span {
    font-size: 0.5rem;
}

.description {
    margin-left: 1em;
    line-height: 1rem;
    font-size: 0.8rem;
    margin-bottom: 1em;
}

}








