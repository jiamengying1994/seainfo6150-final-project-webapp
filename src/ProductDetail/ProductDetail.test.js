import React from "react";
import { render } from "@testing-library/react";
import ProductDetail from "./ProductDetail.jsx";
import { BrowserRouter, Route, Link } from 'react-router-dom';

describe("ProductDetail test", () => {
  it("renders correctly", () => {
    const al = [{"title": "1","image": "1"},{"title": "2","image": "2"}]
    const { container} = render (
    <BrowserRouter>
      <ProductDetail product={al}/>
    </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});