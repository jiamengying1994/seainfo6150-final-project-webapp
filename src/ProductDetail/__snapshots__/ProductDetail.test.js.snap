// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetail test renders correctly 1`] = `
<div>
  <div
    class="ProductDetail"
  >
    <div
      class="detailbox"
    >
      <div
        class="detailImage"
      >
        <img />
      </div>
      <div
        class="detailList"
      >
        <h1>
           
           
        </h1>
        <ul>
          <li>
            <span />
            <p>
              Category
            </p>
          </li>
          <li>
            <span />
            <p>
              Dimensions
            </p>
          </li>
          <li>
            <span />
            <p>
              Techniques
            </p>
          </li>
        </ul>
        <div
          class="description"
        />
      </div>
    </div>
  </div>
</div>
`;
